client/implementation-vertx/src/main/java/io/smallrye/graphql/client/vertx/dynamic/VertxDynamicGraphQLClient.java

```diff
@@ -55,15 +55,20 @@
     // and a new websocket connection attempted.
     private final AtomicReference<Uni<WebSocketSubprotocolHandler>> webSocketHandler = new AtomicReference<>();
 
-    VertxDynamicGraphQLClient(Vertx vertx, String url, String websocketUrl, boolean executeSingleOperationsOverWebsocket,
+    VertxDynamicGraphQLClient(Vertx vertx, WebClient webClient,
+            String url, String websocketUrl, boolean executeSingleOperationsOverWebsocket,
             MultiMap headers, WebClientOptions options,
             List<WebsocketSubprotocol> subprotocols, Integer subscriptionInitializationTimeout) {
         if (options != null) {
```

Comment:
As noted in my other comment, my only concern here is the potential consequences of having two different underlying instances of HttpClient, one from the explicitly-passed WebClient, and one being constructed here from vertx. I do see, though, that there has been a recent enhancement which closes both the HttpClient and the WebClient separately, which is good!